Wait fixes

Kira-Vishn9 opened the pull request.

Commit:
feat: init vite react and mui

Commit:
feat: LogIn aside header

linkmss committed:
dev: Eslint. WIP

linkmss committed:
dev: Add make

Commit:
pass

Commit:
Merge branch 'dev' of https://github.com/Kira-Vishn9/AdminAudio into dev

Commit:
pull

Commit:
pull

Kira-Vishn9 committed:
add all change

Kira-Vishn9 committed:
add all change

Commit:
refactoring all error need help with button in LogIn

linkmss committed:
dev: Add alias

Kira-Vishn9 committed:
done auth

Kira-Vishn9 committed:
help

linkmss committed:
dev: Add root layouts

Kira-Vishn9 committed:
add something api and card to main page

Kira-Vishn9 committed:
add type dor response

linkmss committed:
dev: Modules. Base

Kira-Vishn9 committed:
refactor code and add get resp with moc data

linkmss committed:
dev: Modules. WIP

Kira-Vishn9 committed:
fix mistake with type

linkmss committed:
dev: Modules. WIP

Kira-Vishn9 committed:
add almost type

Kira-Vishn9 committed:
fix: refact style

Kira-Vishn9 committed:
feat: new page newsong

linkmss committed:
dev: Add song page with params

Kira-Vishn9 committed:
feat: new route

Kira-Vishn9 committed:
fix: refact singlesong

linkmss committed:
dev: Add todo and other

Kira-Vishn9 committed:
feat: add input toke

Kira-Vishn9 committed:
fix: refact code

linkmss commented on react/tsconfig.json:
```diff
@@ -0,0 +1,35 @@
+      ],
+      "@components/*": [
+        "./src/component/*"
+      ],
```
добавить элиасы

Kira-Vishn9 replied:
done

linkmss commented on react/index.html:
```diff
@@ -0,0 +1,13 @@
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
привести в порядок: поменять иконку, тайтл

Kira-Vishn9 replied:
done

linkmss commented on react/src/index.css:
```diff
@@ -0,0 +1,27 @@
+:root {
```
Удалить app.css
Добавить normilize.css + в index.css -> index.scss
index.scss { @import 'normilize' }

Kira-Vishn9 replied:
done

linkmss commented on react/src/type/allType.ts:
```diff
@@ -0,0 +1,19 @@
+  children: JSX.Element | JSX.Element[]
+}
+
+export interface IArtist {
```
del

Kira-Vishn9 replied:
done

linkmss commented on react/src/type/allType.ts:
```diff
@@ -0,0 +1,19 @@
+export interface Ichild {
```
IComponentChild

Kira-Vishn9 replied:
done

linkmss commented on react/src/router/router.ts:
```diff
@@ -0,0 +1 @@
+
```
del

linkmss replied:
coversation: Del router.ts. WIP - work in progress
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923682

Commit example

Kira-Vishn9 replied:
done

linkmss commented on react/public/vite.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="iconify iconify--logos" width="31.88" height="32" preserveAspectRatio="xMidYMid meet" viewBox="0 0 256 257"><defs><linearGradient id="IconifyId1813088fe1fbc01fb466" x1="-.828%" x2="57.636%" y1="7.652%" y2="78.411%"><stop offset="0%" stop-color="#41D1FF"></stop><stop offset="100% ...
```
заменить на брендовую иконку

Kira-Vishn9 replied:
done

linkmss commented on react/src/api/index.ts:
```diff
@@ -0,0 +1,9 @@
+
+export { type AxiosResponse } from 'axios'
+export const httpClient = Axios.create({
+  baseURL: 'http://54.36.100.210:3712/api',
```
подключить в env BASE_URL

Kira-Vishn9 replied:
also need help

linkmss commented on react/src/api/index.ts:
```diff
@@ -0,0 +1,9 @@
+  withCredentials: true
+})
+
+httpClient.interceptors.response.use((axiosResponse) => axiosResponse.data)
```
через httpClient.interceptors.request.use - добавить кастомный заголовок

Kira-Vishn9 replied:
need help

linkmss commented on react/src/component/Aside/Aside.tsx:
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { Apps, List, ListItem, ListItemButton, ListItemDecorator } from '@components/index'
+import './style.css'
```
везде на модули

Kira-Vishn9 replied:
done

linkmss commented on react/src/component/Header/Header.tsx:
```diff
@@ -0,0 +1,26 @@
+import { Button } from '@components/index'
+import { Link } from 'react-router-dom'
+import React from 'react'
+import logo from '@/assets/logo.svg'
```
ауф контекст - часть модуля ауторизации и должен экспортироваться из index этого модуля

Kira-Vishn9 replied:
check

linkmss commented on react/src/component/Layout/LauoutAuth.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Header } from '../Header/Header'
```
Во всем проекте ../ и прочее заменить на элиасы через @*

Kira-Vishn9 replied:
done

linkmss commented on react/src/component/Layout/LauoutAuth.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Header } from '../Header/Header'
+import { Aside } from '../Aside/Aside'
```
LayoutAuth -> LayoutDefault. Обычно layoutAuth - это лейаут для авторизации. Соответственно default - лейаут авторизованного юзера

Kira-Vishn9 replied:
done

linkmss commented on react/src/component/index.ts:
```diff
@@ -0,0 +1,22 @@
+import { Button } from '@mui/material'
+import List from '@mui/joy/List'
```
если дефолтн импорт -> export * as List from '@mui/joit/List

если нет -> export { Button } from '@mui/material'

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/login.service.ts:
```diff
@@ -0,0 +1,8 @@
+
+// TODO:
+// 1. хранить токен
```
ну соответственно сделать

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+    mail && password && token ? setAuth(true) : alert('try again')
+  }
+
+  const validationMail = (e: React.ChangeEvent<HTMLInputElement>): undefined | boolean => {
```
если функция валидации, которая еще и возвращает валидна почта или нет - getIsEmailValid - лучше будет название
Так для всех функций-валидаторов

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+    mail && password && token ? setAuth(true) : alert('try again')
+  }
+
+  const validationMail = (e: React.ChangeEvent<HTMLInputElement>): undefined | boolean => {
```
можно сделать возвращаемый функцией тип только булеан, если if (target === null) return false будет

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+        setToken(res.status)
+      })
+      .catch((err) => {
+        console.log(err)
```
На все консоль еррор добавить тост с ошибкой из mui

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+        <img src={logo}/>
+      </div>
+      <Input
+        style={{ marginTop: '10px', marginBottom: '10px' }}
```
Убрать инлайновые стили

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+        onClick={checkin}
+      >
+          <Link to={'/songs'} >
+            Enter
```
sign up

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/login/LogIn.tsx:
```diff
@@ -0,0 +1,87 @@
+  const [istoken, setIsToken] = React.useState(false)
+
+  const checkin = (): void => {
+    mail && password && token ? setAuth(true) : alert('try again')
```
асинк чекин, 1 этап - валидация всего, валидны - идем дальше. нет - тост с ошибкой

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/BackButton/BackButton.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
```
вынести в общие компоненты

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/BackButton/BackButton.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
```
Вынести в общие компоненты

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
```
Пагинацию тоже в общие компоннеты

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/RoundedDiv/RoundedDiv.tsx:
```diff
@@ -0,0 +1,42 @@
+  imgSrc: string
+}
+export const RoundedDiv = ({ songInfo, addCover, imgSrc }: IRoundedDivProps): JSX.Element => {
+  const divStyle = {
```
убрать в стили

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/RoundedDiv/RoundedDiv.tsx:
```diff
@@ -0,0 +1,42 @@
+          )
+        : (
+          <label htmlFor="imgLoader" style={{ display: 'flex', justifyContent: 'center', alignSelf: 'center', cursor: 'pointer' }}>
+            <div style={divStyle}>{songInfo !== undefined ? songInfo.name.split(' ').slice(1, 2).join(' ') : 'init'}</div>
```
songInfo !== undefined ? songInfo.name.split(' ').slice(1, 2).join(' ') : 'init' - вынести в функцию getSomething

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/RoundedDiv/RoundedDiv.tsx:
```diff
@@ -0,0 +1,42 @@
+          )
+        : (
+          <label htmlFor="imgLoader" style={{ display: 'flex', justifyContent: 'center', alignSelf: 'center', cursor: 'pointer' }}>
+            <div style={divStyle}>{songInfo !== undefined ? songInfo.name.split(' ').slice(1, 2).join(' ') : 'init'}</div>
```
Правила

1. В верске не используются вычисления вида songInfo !== undefined ? songInfo.name.split(' ').slice(1, 2).join(' ') : 'init' - они заменяются функциям getName

правила именований
isActive - boolean

getIsValid - () => boolean
isValid - boolean
onSomething - хендлер для событий

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/Item.tsx:
```diff
@@ -0,0 +1,35 @@
+        </TableCell>
+        <TableCell align="right">{(new Date(songs.release_date).toLocaleDateString())}</TableCell>
+        <TableCell align="right">
+          <Link to={`/song/${songs._id}`}>
```
Проверить весь проект на наличие конструкции
Button with onClick -> link to

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/components/SingleSong.tsx:
```diff
@@ -0,0 +1,102 @@
+            </label>
+            <label>
+              Genres:
+              <Input sx={{ width: '100%' }} {...register('genres')} disabled={true} defaultValue={songInfo.genres.map(el => el.name).join(',')} />
```
songInfo.genres.map(el => el.name).join(',') - function во всем проекте

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/views/Song.tsx:
```diff
@@ -0,0 +1,46 @@
+const Song = (): JSX.Element => {
+  const { id } = useParams()
+  const [selectedSong, setSelectedSong] = React.useState<ISongFilled | undefined>()
+  const [isLoad, setIsLoad] = React.useState(true)
```
привыкай к флагу isFetching

Kira-Vishn9 replied:
привык

linkmss commented on react/src/module/song/views/Song.tsx:
```diff
@@ -0,0 +1,46 @@
+const Song = (): JSX.Element => {
+  const { id } = useParams()
+  const [selectedSong, setSelectedSong] = React.useState<ISongFilled | undefined>()
+  const [isLoad, setIsLoad] = React.useState(true)
```
Как разрулить состояния

1. Сначала пытаешься показать loading - if (isFetching)
2. ELSE - если есть данные - показываем данные
3. Если данных нет - показываем notFound

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/views/Song.tsx:
```diff
@@ -0,0 +1,46 @@
+  const [selectedSong, setSelectedSong] = React.useState<ISongFilled | undefined>()
+  const [isLoad, setIsLoad] = React.useState(true)
+
+  React.useEffect(() => {
```
SONGS.tsx зарефакторить

Kira-Vishn9 replied:
done

linkmss commented on react/src/module/song/index.ts:
```diff
@@ -0,0 +1,7 @@
+// page
+import { Songs } from './views/Songs.tsx'
+
+export { type ISong, getSongs, Songs, type IAlbums }
```
мы наружу getSongs не отдаем

Kira-Vishn9 replied:
done

linkmss commented on react/src/App.tsx:
```diff
@@ -0,0 +1,41 @@
+import { Songs } from './module/song'
+
+import React from 'react'
+import AddSong from '@/module/song/views/AddSong.tsx'
```
Убрать и сделать через index

Kira-Vishn9 replied:
done

linkmss commented on react/src/App.tsx:
```diff
@@ -0,0 +1,41 @@
+import { Songs } from './module/song'
+
+import React from 'react'
+import AddSong from '@/module/song/views/AddSong.tsx'
```
CreateSong

Kira-Vishn9 replied:
done

linkmss reviewed (changes requested):
Review#1

Kira-Vishn9 committed:
fix: change name AddSong to CreateSong https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252974568

Kira-Vishn9 committed:
fix: delete superfluous import and export https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252971874

Kira-Vishn9 committed:
fix: export only via index https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252974204

Kira-Vishn9 committed:
fix: fix state https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252969024

Kira-Vishn9 committed:
fix: refact songs https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252970846

Kira-Vishn9 committed:
fix: refactoring isLoading to isFetching https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252967567

Kira-Vishn9 committed:
fix: refactoring singleSong https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252965761

Kira-Vishn9 committed:
fix: refactoring delet link in button https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252964799

Kira-Vishn9 committed:
fix: refactoring https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923309
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252922963
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252922555
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252919723
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923682
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923982
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252927076
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252933164
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252950935
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252951361
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252959640

Kira-Vishn9 committed:
fix: refactoring https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923309
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252922963
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252922555
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252919723
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923682
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252923982
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252927076
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252933164
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252950935
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252951361
https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252959640

Kira-Vishn9 committed:
fix: fix import context

Kira-Vishn9 committed:
feat: add normalize
 https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252922555

Kira-Vishn9 committed:
feat: new alias in tsconfig
 https://github.com/Kira-Vishn9/AdminAudio/pull/2#discussion_r1252918029

Kira-Vishn9 committed:
feat: add modal window

linkmss committed:
dev: Add modalLayer

Kira-Vishn9 committed:
feat: add new modal window

Kira-Vishn9 committed:
feat: add new modal window genres
